src/cow-react/modules/swap/containers/EthFlow/EthFlowBanner.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react'
+import { EthFlowBannerContent } from '@cow/modules/swap/pure/EthFlow/EthFlowBanner'
+import { useDetectNativeToken } from '@src/custom/state/swap/hooks'
+import { useHasEnoughWrappedBalanceForSwap } from '@src/custom/hooks/useWrapCallback'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+export interface EthFlowBannerCallbacks {
+  wrapCallback: () => void
+  forceWrapCallback: () => void
+  switchCurrencyCallback: () => void
+}
+
+export interface EthFlowBannerProps extends EthFlowBannerCallbacks {
+  nativeInAmount: CurrencyAmount<Currency> | undefined
+}
+
+export default function EthFlowBanner({ nativeInAmount, ...props }: EthFlowBannerProps) {
+  const [showBanner, setShowBanner] = useState(false)
+  const hasEnoughWrappedBalance = useHasEnoughWrappedBalanceForSwap(nativeInAmount)
+  const { isNativeIn, native, wrappedToken: wrapped } = useDetectNativeToken()
+
+  const showBannerCallback = () => {
+    return setShowBanner((state) => !state)
+  }
+
+  // dont render if it isn't a native token swap
+  if (!isNativeIn) return null
```

Comment:
nit: If it's wrapping, then we should also return null, right? 
I mean it will anyways not be shown, but I guess its good practice?

Reply:
isNativeIn is implicitly checking this, all good :)